File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        res = []
        l = 0
        r = len(heights)-1

        while l<r:
            a = heights[l]
            b = heights[r]

            if a > b:
                res.append(b*(r-l))
                r -= 1
            elif b>a:
                res.append(a*(r-l))
                l+=1
            else:
                res.append(a*(r-l))
                r -= 1
        
        return max(res)
        

File: Data Structures & Algorithms/fruit-into-baskets/submission-0.cpp
class Solution {
public:
    int totalFruit(vector<int>& fruits) {
        unordered_map<int, int> map;
        int l = 0 , total = 0 , res = 0;

        for(int r = 0 ; r < fruits.size() ; r++){
            total++;
            map[fruits[r]]++;

            while(map.size() > 2){
                int k = fruits[l];
                map[k]--;
                total--;
                if(map[k] == 0){
                    map.erase(k);
                }
                l++;

            }
            res = max(res , total);
        }
        return res;
    }
};